### PR TITLE
Add Openpose from TF

### PR DESCRIPTION
**Pull request by jkamelin**

### CODE REVIEW

File: tools/accuracy_checker/configs/openpose-cmu-tf.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            aspect_ratio_scale: width
+          - type: padding
+            stride: 8
+
+
+    launchers:
+      - framework: dlsdk
+        tags:
+          - FP32
+        model:   public/openpose-cmu-tf/FP32/opt_graph.bin
```
Comment (eaidova) [inline tools/accuracy_checker/configs/openpose-cmu-tf.yml]: I am not sure about converted model name correctness... As I see it build based on model directory name.
so possibly it is
`public/openpose-cmu-tf/FP32/openpose-cmu-tf.xml` ?

Comment (eaidova) [reply tools/accuracy_checker/configs/openpose-cmu-tf.yml]: yes, it is true, look on model downloader precommit logs:
Conversion command: venv/conversion/bin/python -- /opt/intel/openvino/deployment_tools/model_optimizer/mo.py --framework=tf --data_type=FP32 --output_dir=download/public/openpose-cmu-tf/FP32 --model_name=openpose-cmu-tf '--input_shape=[1,368,432,3]' --input=image --output=Mconv7_stage6_L2/Conv2D,Mconv7_stage6_L1/Conv2D --input_model=download/public/openpose-cmu-tf/graph_opt.pb

File: tools/accuracy_checker/configs/openpose-cmu-tf.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+models:
+  - name: openpose-cmu-tf
+
+    launchers:
+      - framework: tf
+        model: public/openpose-cmu-tf/opt_graph.pb
+        adapter:
+          type: human_pose_estimation
+          part_affinity_fields_out: Mconv7_stage6_L1/Conv2D
```
Comment (eaidova) [inline tools/accuracy_checker/configs/openpose-cmu-tf.yml]: Is it not required to specify outputs explicitly?
As I remember, it has concat of this 2 layers as output.

File: tools/accuracy_checker/configs/openpose-cmu-tf.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          keypoints_heatmap_out: Mconv7_stage6_L2/Conv2D
+
+    datasets:
```
Comment (eaidova) [inline tools/accuracy_checker/configs/openpose-cmu-tf.yml]: if original framework and IR does not have difference in preprocessing, no need to specify source framework as separated model, you can have only one model specification with 3 launchers (tf, dlsdk with FP32 tag and dlsdk with FP16 tag)

File: models/public/openpose-cmu-tf/openpose-cmu-tf.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+- `W` - width
+- `C` - heatmap channel
+
+### Converted model
+
+1. Predicted keypoint heatmaps, name - `Mconv7_stage6_L2/Conv2D`, shape - `1,19,46,54`,
+   output data format is `B,C,H,W` where:
+
+- `B` - batch size
+- `C` - number of predicted body parts according to MSCOCO kepoints classification
```
Comment (eaidova) [inline models/public/openpose-cmu-tf/openpose-cmu-tf.md]: it is not correct:
* `MSCOCO keypoints` challenge works only with 17 keypoints (no keypoint for neck, the last for backgroud):
0 | "nose"
1 | "left_eye"
2 | "right_eye"
3 | "left_ear"
4 | "right_ear"
5 | "left_shoulder"
6 | "right_shoulder"
7 | "left_elbow"
8 | "right_elbow"
9 | "left_wrist"
10 | "right_wrist"
11 | "left_hip"
12 | "right_hip"
13 | "left_knee"
14 | "right_knee"
15 | "left_ankle"
16 | "right_ankle"
* the keypoints order is not equvivalent to COCO
see: https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/output.md#pose-output-format-coco

Comment (eaidova) [reply models/public/openpose-cmu-tf/openpose-cmu-tf.md]: for better understanding, [mapping keypoints to coco order in AC](https://github.com/opencv/open_model_zoo/blob/master/tools/accuracy_checker/accuracy_checker/annotation_converters/cvat_human_pose.py#L23-L41)

File: models/public/openpose-cmu-tf/openpose-cmu-tf.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+name - `Openpose/concat_stage7`, output data format is `B,H,W,C` where:
+
+- `B` - batch size
+- `H` - height
+- `W` - width
+- `C` - heatmap channel
+
+### Converted model
+
+1. Predicted keypoint heatmaps, name - `Mconv7_stage6_L2/Conv2D`, shape - `1,19,46,54`,
```
Comment (eaidova) [inline models/public/openpose-cmu-tf/openpose-cmu-tf.md]: I think, it is more logical to specify outputs in topological order, so please change places for `Mconv7_stage6_L2/Conv2D` and `Mconv7_stage6_L1/Conv2D`

File: models/public/index.md [hunk 6]
```diff
@@ -123,6 +123,21 @@ SSD-based and provide reasonable accuracy/performance trade-offs.
...
+video.  Keypoints are body joints, i.e. ears, eyes, nose, shoulders, knees, etc.
+There are two major groups of such metods: top-down and bottom-up.  The first
+detects persons in a given frame, crops or rescales detections, then runs pose
+estimation network for every detection. These methods are very accurate. The
+second finds all keypoints in a given frame, then groups them by person
+instances, thus faster than previous, because network runs once.
+
+| Model Name                | Implementation | OMZ Model Name | Accuracy | GFlops | mParams |
+| ------------------------- | ---------------| -------------- | -------- | ------ | ------- |
+| OpenPose                  | [TensorFlow\*](./openpose-cmu-tf/openpose-cmu-tf.md) | openpose-cmu-tf | 49.89 | 319.304 | 52.311 |
```
Comment (eaidova) [inline models/public/index.md]: why do you set accuracy in index.md and stay this field empty in the model config?
Is it made on some reason?

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: models/public/openpose-cmu-tf/model.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  This is TensoFlow version of OpenPose algorithm, used for multi-person
+  human pose estimation. The model is based on VGG pretrained network,
+  which described in the original paper <https://arxiv.org/pdf/1812.08008.pdf>.
+  For details see repository <https://github.com/ildoonet/tf-pose-estimation>.
+task_type: human_pose_estimation
+files:
+  - name: graph_opt.pb
+    size: 209299198
+    sha256: 1fc81fff68d2351dba0bf206ae4431a760a202151ff539feaefdaab534d79aeb
+    source: http://download686.mediafire.com/xk527xyl3r6g/qlzzr20mpocnpa3/graph_opt.pb
```
Comment (IRDonch) [inline models/public/openpose-cmu-tf/model.yml]: This link no longer works (well, it does, but it doesn't directly point to the file anymore).

There is clearly no way to get a permanent download link from MediaFire. We could, in principle, add a dedicated source type for MediaFire, like we do for Google Drive, and try to screen scrape the download link from the MediaFire webpage, but I don't think we should. There is no way to do this reliably, as MediaFire could alter its design at any moment, or even put up a CAPTCHA.

I think we'll just have to write this model off as a lost cause.

Comment (jkamelin) [reply models/public/openpose-cmu-tf/model.yml]: Yes, there is no direct link to the model. There is also no respond from authors on our request to place model somewhere else. @vladimir-dudnik, are we going to redistribute this model through another storage (download01.org?) or exclude it?

**Pull request closed without merging**